src/api/openThesaurusAPI.ts

```diff
@@ -0,0 +1,51 @@
+import { log } from "console";
+import type { PartOfSpeech, Synonym, SynonymProvider } from "src/api/types";
+
+class Base {
+  API_END_POINT: string = "https://www.openthesaurus.de/synonyme/search?q=";
+
+  public name: string = "OpenThesaurus";
+  public url: string = "https://www.openthesaurus.de/";
+
+  /**
+   * @param query - The term you want to look up
+   * @returns Returns the URL in REST schema
+   */
+  constructRequest(query: string): string {
+    return this.API_END_POINT + query + "&format=application/json";
+    //SCHEMA: https://www.openthesaurus.de/synonyme/search?q=<QUERY>&format=application/json
+  }
+}
+
+export class openThesaurusAPIProvider extends Base implements SynonymProvider {
```

Comment:
After merging this Class with the `Base` Class, could you please rename it to `openThesaurusSynonymAPI`, so that it clearly states what it is.